test/parallel/test-tick-processor.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+var fs = require('fs');
+var assert = require('assert');
+var path = require('path');
+var cp = require('child_process');
+var common = require('../common');
+
+common.refreshTmpDir();
+
+process.chdir(common.tmpDir);
+cp.execFileSync(process.execPath, ['-prof', '-pe',
+    'function fib(n) { require(\'vm\').runInDebugContext(\'Debug\');' +
+    'return n < 2 ? n : setImmediate(function() { fib(n-1) + fib(n-2);}); };' +
+    'setTimeout(function() { process.exit(0); }, 2000);' +
+    'fib(40);']);
```

Comment:
I think `setTimeout` does work because of the intentional `setImmediate` inside `fib`.